src/components/DropZone/DropZone.tsx

```diff
@@ -224,26 +230,27 @@ export class DropZone extends React.Component<CombinedProps, State> {
       styles.DropZone,
       outline && styles.hasOutline,
       (active || dragging) && styles.isDragging,
+      loading && styles.isLoading,
       error && styles.hasError,
-      size && size === 'extraLarge' && styles.sizeExtraLarge,
-      size && size === 'large' && styles.sizeLarge,
-      size && size === 'medium' && styles.sizeMedium,
-      size && size === 'small' && styles.sizeSmall,
+      height && height === Size.ExtraLarge && styles.sizeExtraLarge,
+      height && height === Size.Large && styles.sizeLarge,
+      height && height === Size.Medium && styles.sizeMedium,
+      height && height === Size.Small && styles.sizeSmall,
     );
 
     const dragOverlay =
       (active || dragging) && !error && overlay ? (
         <div className={styles.Overlay}>
           <Stack vertical spacing="tight">
             <Icon source={IconDragDrop} color="indigo" />
-            {size === 'extraLarge' && (
-              <DisplayText size="small" element="p">
-                {overlayText}
-              </DisplayText>
-            )}
-            {(size === 'medium' || size === 'large') && (
-              <Caption>{overlayText}</Caption>
-            )}
+            {width === Size.ExtraLarge &&
```

Comment:
not necessary but I would have probably pulled this logic out into it's own variable.

Reply:
Maybe a switch statement to determine the markup rendered?

Reply:
I removed the variable sections out. I think a switch statement would repeat to much logic. I made the caption its own conditional variable and the Display Text its own conditional variable since the Icon itself will always render.